goinstant/docker_test.go

```diff
@@ -0,0 +1,948 @@
+		startupCommands []string
+		expectedResults resultStruct
+		testInfo        string
+	}{
+		{
+			startupCommands: []string{"init", "-t=docker", "--instant-version=v2.0.1", "-c=../test", "-c=../test1", "-e=NODE_ENV=dev", "-onlyFlag", "core"},
+			expectedResults: resultStruct{
+				environmentVariables: []string{"-e", "NODE_ENV=dev"},
+				deployCommand:        "init",
+				otherFlags:           []string{"-onlyFlag"},
+				targetLauncher:       "docker",
+				packages:             []string{"core"},
+				customPackagePaths:   []string{"../test", "../test1"},
+				instantVersion:       "v2.0.1",
+			},
+			testInfo: "Extract commands test 1 - should return the expected commands",
+		},
+		{
+			startupCommands: []string{"up", "-t=kubernetes", "--instant-version=v2.0.2", "-c=../test", "-c=../test1", "-e=NODE_ENV=dev", "-onlyFlag", "core"},
+			expectedResults: resultStruct{
+				environmentVariables: []string{"-e", "NODE_ENV=dev"},
+				deployCommand:        "up",
+				otherFlags:           []string{"-onlyFlag"},
+				targetLauncher:       "kubernetes",
+				packages:             []string{"core"},
+				customPackagePaths:   []string{"../test", "../test1"},
+				instantVersion:       "v2.0.2",
+			},
+			testInfo: "Extract commands test 2 - should return the expected commands",
+		},
+		{
+			startupCommands: []string{"down", "-t=k8s", "--instant-version=v2.0.2", "-c=../test", "-c=../test1", "--env-file=../test.env", "-onlyFlag", "core", "hapi-fhir"},
+			expectedResults: resultStruct{
+				environmentVariables: []string{"--env-file", "../test.env"},
+				deployCommand:        "down",
+				otherFlags:           []string{"-onlyFlag"},
+				targetLauncher:       "k8s",
+				packages:             []string{"core", "hapi-fhir"},
+				customPackagePaths:   []string{"../test", "../test1"},
+				instantVersion:       "v2.0.2",
+			},
+			testInfo: "Extract commands test 3 - should return the expected commands",
+		},
+		{
+			startupCommands: []string{"destroy", "-t=swarm", "--instant-version=v2.0.2", "--custom-package=../test", "-c=../test1", "-e=NODE_ENV=dev", "--onlyFlag", "core", "hapi-fhir"},
+			expectedResults: resultStruct{
+				environmentVariables: []string{"-e", "NODE_ENV=dev"},
+				deployCommand:        "destroy",
+				otherFlags:           []string{"--onlyFlag"},
+				targetLauncher:       "swarm",
+				packages:             []string{"core", "hapi-fhir"},
+				customPackagePaths:   []string{"../test", "../test1"},
+				instantVersion:       "v2.0.2",
+			},
+			testInfo: "Extract commands test 4 - should return the expected commands",
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.testInfo, func(t *testing.T) {
+			environmentVariables, deployCommand, otherFlags, packages, customPackagePaths, instantVersion, targetLauncher := extractCommands(tt.startupCommands)
+
+			if !assert.Equal(t, environmentVariables, tt.expectedResults.environmentVariables) {
+				t.Fatal("ExtractCommands should return the correct environment variables")
```

Comment:
See later comment about t.Fatal(). Not going to mention that at every instance, but think about whether or not we want all test cases to quit if we fail one test case.